File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/adapter/shop/ui/client/pages/HomePage.java
package com.optivem.shop.dsl.driver.adapter.shop.ui.client.pages;

import com.optivem.shop.dsl.driver.adapter.shared.client.playwright.PageClient;

public class HomePage extends BasePage {
    private static final String SHOP_BUTTON_SELECTOR = "a[href='/shop']";
    private static final String ORDER_HISTORY_BUTTON_SELECTOR = "a[href='/order-history']";
    public HomePage(PageClient pageClient) {
        super(pageClient);
    }

    public NewOrderPage clickNewOrder() {
        pageClient.click(SHOP_BUTTON_SELECTOR);
        return new NewOrderPage(pageClient);
    }

    public OrderHistoryPage clickOrderHistory() {
        pageClient.click(ORDER_HISTORY_BUTTON_SELECTOR);
        return new OrderHistoryPage(pageClient);
    }
}

File: system-test/dotnet/Driver.Adapter/Shop/Ui/Client/Pages/HomePage.cs
using Driver.Adapter.Shared.Client.Http;

using Driver.Adapter.Shared.Client.Playwright;



namespace Driver.Adapter.Shop.Ui.Client.Pages;



public class HomePage : BasePage

{

    private const string ShopButtonSelector = "a[href='/new-order']";

    private const string OrderHistoryButtonSelector = "a[href='/order-history']";



    public HomePage(PageClient pageClient) : base(pageClient)

    {

    }



    public async Task<NewOrderPage> ClickNewOrderAsync()

    {

        await PageClient.ClickAsync(ShopButtonSelector);

        return new NewOrderPage(PageClient);

    }



    public async Task<OrderHistoryPage> ClickOrderHistoryAsync()

    {

        await PageClient.ClickAsync(OrderHistoryButtonSelector);

        return new OrderHistoryPage(PageClient);

    }



}





